src/core/tests/widgets/test_icon.py

```diff
@@ -9,4 +10,10 @@ def setUp(self):
         self.factory = MagicMock()
         self.factory.Icon = MagicMock(return_value=MagicMock(spec=toga_dummy.factory.Icon))
 
-        self.icon = toga.Icon(self.text, factory=self.factory)
\ No newline at end of file
+        self.test_path = "Example.bmp"
+        self.icon = toga.Icon(self.test_path)
+
+    def test_icon_bind(self):
+        self.assertEqual(self.icon._impl, None)
+        self.icon.bind(factory=toga_dummy.factory)
+        self.assertEqual(self.icon._impl.interface, self.icon)
```

Comment:
It would also be a good idea to check that the bound icon is actually the same one that was passed in.